dlrover/python/elastic_agent/torch/training.py

```diff
@@ -946,14 +952,14 @@ def _stop_workers(
                     super()._stop_workers(worker_group, is_restart)
 
             signal.alarm(0)
-        except TimeoutError as te:
+        except RendezvousTimeoutError as te:
             logger.error(str(te))
-            raise
+            raise te
         finally:
             signal.alarm(0)
 
     def _stop_timeout_handler(self, signum, frame):
-        raise TimeoutError("Timed out waiting for stopping workers.")
+        raise RendezvousTimeoutError("Timed out waiting for stopping workers.")
```

Comment:
The issue still persists: the timeout here for 'stop_workers' is unrelated to the rendezvous (rdzv).